os_migrate/plugins/module_utils/workload_common.py

```diff
@@ -28,54 +28,22 @@
     DEVNULL = open(os.devnull, 'r+', encoding='utf8')
 
 
-def check_process(pid):
-    """Check whether pid exists in the current process table."""
-    if pid is None:
-        return False
-    try:
-        os.kill(int(pid), 0)
-    except (OSError, ValueError):
-        return False
-    else:
-        return True
-
-
-def check_and_cleanup_lockfiles():
-    """
-    Removes lockfiles found in specific directories.
-    If a lockfile is not being used by a process,
-    it is either deleted or not being used.
-    """
-    # Remove specific lockfiles
-    for lockfile in [ATTACH_LOCK_FILE_SOURCE, ATTACH_LOCK_FILE_DESTINATION, PORT_LOCK_FILE]:
-        try:
-            with open(f"{lockfile}.lock", "r", encoding='utf-8') as f:
-                pid = f.read().strip()
-            if check_process(pid):
-                # print(f"Lockfile {lockfile} is currently in use by process: {pid}")
-                pass
-            else:
-                os.remove(lockfile + '.lock')
-                return True
-        except FileNotFoundError:
-            return False
-
-
 def use_lock(lock_file):
     """ Boilerplate for functions that need to take a lock. """
     def _decorate_lock(function):
         def wait_for_lock(self):
-            if check_and_cleanup_lockfiles():
+            if self.check_and_cleanup_lockfiles():
                 self.log.info("Cleaned up unused lockfiles before start")
             for second in range(DEFAULT_TIMEOUT):
                 try:
                     self.log.info('Waiting for lock %s...', lock_file)
+                    time.sleep(1)
```

Comment:
This will wait always, but we only need to wait if the first attempt fails and we need to make another attempt. I think the right spot for this waiting is right before the line `except subprocess.CalledProcessError as err`. (It should not be a part of the `elif result == 0` block though.)

Reply:
I thought we wanted to place the time.sleep before line 86?